fix: return module

pintsized opened the pull request:
Previously loading this module would set `pack` and `unpack` into the global `string` library. Since these methods now exist (since Lua 5.3), other Lua modules will use these functions if they exist, assuming that they are the implementations in Lua 5.3.

The upshot being that in various places where this module is used, the user is forced to unload it and reset the string methods in order to avoid breaking other code.

More generally, modern best practice is to return the module when loading instead of setting anything globally.

Obviously this is a breaking change, and would require a major version bump.

Commit:
fix: return module

Previously loading this module would set `pack` and `unpack` into the
global `string` library. Since these methods now exist (since Lua 5.3),
other Lua modules will use these functions if they exist, assuming that
they are the implemenations in Lua 5.3.

More generally, modern best practice is to return the module when
loading instead of setting anything globally.

Obviously this is a breaking change.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kong/lua-pack?pullRequest=3) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Kong/lua-pack?pullRequest=3) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kong/lua-pack?pullRequest=3) it.</sub>

Tieske reviewed (approved):
lgtm

javierguerragiraldez committed:
bump version to 2.0.0

javierguerragiraldez committed:
Remove Makefile

Use `luarocks build` or `luarocks make` instead.  The "builtin" process
is much better at providing the appropriate CFLAGS.

javierguerragiraldez committed:
Fix test

Still not a real test, since it doesn't show failure, but at least now
it runs to completion.

javierguerragiraldez committed:
Update README.md

Since it no longer pollutes the `string` table, it's just a normal
module.  Also use a bit of formatting.
